### DISCUSSION TITLE
How to free up disk space when it is less than minFreeSpacePercent

### DISCUSSION THREAD

**Discussion by wusongANKANG** [seaweedfs/seaweedfs#5716] [Q&A]
When the disk space is less than minFreeSpacePercent, the volume server does not allow writing, and all volumes are in read-only state. Vacuum will not be executed at this time.

https://github.com/seaweedfs/seaweedfs/blob/cb67137a03173ac3640783dc0e4a9bd871a13b37/weed/topology/topology_vacuum.go#L273

Then even if all data is deleted, the disk space cannot be released.

How to deal with this situation? Should the isReadOnly restriction be removed during garbage collection?

**Comment by Zen3515** [comment]
I can't believe it, I was about to ask the exact same question.

This is the reason that I said that seaweedfs vacuum is not fast enough in #5661
My gut assume that this was the reason, but I was able to confirm it today and about to ask this question.

**Reply by Zen3515** [reply]
My current workaround is to move some readonly volume from one of the disk to another disk while the volume server is stopped. Then run vacuum.

I think it would be nice if there's a command that could balance the volume between different disks within the same volume server.
Otherwise, one could start each disk (or a set of disks) as one volume server, then we could balance between the volume servers there.

But in order to convert to one disk one volume server, I'm not sure if one could just change the command, and the data will be loaded based on the volume.dat file directly or is it just impossible to do.

**Reply by chrislusf** [reply]
> I think it would be nice if there's a command that could balance the volume between different disks within the same volume server.

Good suggestion! Please create an issue and first.
